lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp

```diff
@@ -2354,6 +2354,8 @@ llvm::Triple SwiftASTContext::GetSwiftFriendlyTriple(llvm::Triple triple) {
     if (triple.isOSLinux() &&
         triple.getEnvironment() == llvm::Triple::UnknownEnvironment)
       triple.setEnvironment(llvm::Triple::GNU);
+    if (triple.isOSWindows() && triple.isWindowsMSVCEnvironment())
+      triple.setVendor(llvm::Triple::UnknownVendor);
```

Comment:
Intuitively I would have expected "microsoft" as a vendor for an msvc environment. Why is is unknown?

Reply:
The open source toolchain itself doesn't have an official vendor behind it.  The `unknown` vendor is the default vendor - it is either that there is no official vendor entity or it is unknown (the former really being a degenerate definition of unknown - there is no known entity vending the toolchain).

Reply:
I think that the other bit of confusion here might be the fact that the MSVC environment does not mean that the toolchain is MSVC.  It means that the C runtime is the Microsoft Visual C runtime.  There are alternative environments like `x86_64-unknown-windows-gnu` (MinGW), `x86_64-unknown-windows-cygnus` (Cygnus' CygWin), and `x86_64-unknown-windows-itanium` - a MSVC compatible environment with an Itanium C++ ABI.